src/Worker.Extensions.DurableTask/HttpHeadersConverter.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+// StringValues does not deserialize as you would expect, so we need a custom mechanism
+// for serializing HTTP header collections
+internal class HttpHeadersConverter : JsonConverter<object>
```

Comment:
Does directly declaring the dictionary here not work?

```suggestion
internal class HttpHeadersConverter : JsonConverter<IDictionary<string, StringValues>>
```